### PR TITLE
add support for zh hant

### PR DESCRIPTION
**Pull request by gabrielo91**
## The Problem

- When users set locale=zh_HK in SPB, they expect all languages showed as Chinese(traditional), but the text on the overlay is showing as Chinese(simplified)

- It seems zh_HK is missing from JS. SDK.
Only Simplified Chinese shows up (zh-CH)

    [example here](https://github.com/paypal/paypal-checkout-components/blob/4885982a8eae958d0a6f75bd4e8362a3eb65effc/src/zoid/checkout/content.js#L154)

- [PayPal sdk client](https://github.com/paypal/paypal-sdk-client/blob/41ef52b877c1a228f989695a69dd307f608977f8/src/script.js#L213) splits the query string like `locale=en-US` into `country=US` and `language=en` if we have something like `locale=en-GB` the country will be `GB`, however the language resolves to `en` as well

- Every time we need to show a specific message based on locales, we have objects like this:

    ```javascript

    const greetings = {
        en: "hi",
        es: "hola"
    }
    
    ```

- which we call like this:
  
  ```javascript
  const sayHi = greetings[language]
  ```

-  so now, we need to resolve an specific value in `en` from GB, we want something like:

    ```javascript
    const greetings = {
        en:    "hi",
        es:    "hola"
        en_gb: "hello"
    }
    ```

- due to the way how `sayHi` function is invoked (and there are several examples like that)
  - [sdk constants](https://github.com/paypal/paypal-sdk-constants/pull/55/files)
  - [checkout components](https://github.com/paypal/paypal-checkout-components/pull/1637/files)
  - [identity components](https://github.com/paypal/paypal-identity-components/pull/14)
 
  we need to add the copies as a new language, in spite of it is just a variant of English (in this case)

## Proposed solutions:

The most evident and simplest one (but the least elegant too) is to add logic in [PayPal sdk client](https://github.com/paypal/paypal-sdk-client/blob/41ef52b877c1a228f989695a69dd307f608977f8/src/script.js#L213) and makes that the function `getLocale` evaluates if the query string is in a special list that contains variants, for example:

1. For the sake of this example, if the query string is
   `locale=en-GB`  and such pair `en-GB` is in our special array, the language will resolve to `en-GB` instead of just `en`

2. in this way, we would add the respective string to the locale objects in all the projects, avoiding using variants

```javascript
    const greetings = {
        en:    "hi",
        es:    "hola"
        en_gb: "hello"
    }
```

and our `getLocal()` function will return something like:

```javascript
    // assuming query param = en-GB
    const { language, country} = getLocale(queryParam);
    // language = en-GB
    // country = gb
```

3. This is a guide made for updating languages to Checkout

- https://github.paypal.com/Globalization-R/G11nMetadata/blob/develop/AddLangsToCheckout.md

Another option consists in add an extra variable when locale is parsed, and get, rather than `country` and `language`, an extra option called `variant` , so, the `getLocal()` function would return something like this:

```javascript
    // assuming query param = en-GB
    const { language, country, variant} = getLocale(queryParam);
    // language = en
    // country = gb
    // variant = nz (just an example)
```

however, the challenge with this approach relies on how to continue working with the current objects that return the copies based only on the language and don´t allow variants or an extra way of 'selector':

```javascript
    const greetings = {
        en:      "hi",
        es:      "hola"
        en_gb:   "hello"
        variant: "variant"
    }

    const sayHi = greetings[languageVariant] // how to incorporate the varian here?
```

This proposed solution relies [here](https://github.com/paypal/paypal-sdk-client/pull/79/files)

## Screenshots

Below screenshots were gotten with the following params 

```html
    <script src="https://localhost.paypal.com:8443/sdk/js?client-id=alc_client1&debug=true&components=buttons,messages&enable-funding=venmo&locale=zh_HK"></script>

```

image

image

- The below screenshots were gotten calling directly to `smartcomponentnodeweb` API directly throw the following link

```url
https://localhost.paypal.com:8443/smart/buttons?style.layout=vertical&style.color=gold&style.shape=rect&style.tagline=false&components.0=buttons&components.1=messages&locale.lang=zh_Hant&locale.country=HK&sdkMeta=eyJ1cmwiOiJodHRwczovL2xvY2FsaG9zdC5wYXlwYWwuY29tOjg0NDMvc2RrL2pzP2NsaWVudC1pZD1hbGNfY2xpZW50MSZkZWJ1Zz10cnVlJmNvbXBvbmVudHM9YnV0dG9ucyxtZXNzYWdlcyZlbmFibGUtZnVuZGluZz12ZW5tbyZsb2NhbGU9emhfSEsiLCJhdHRycyI6eyJkYXRhLXVpZCI6InVpZF92dndlanhraWJwb2dtaWFpc21udnRuaHZxZWpyZnYifX0&clientID=alc_client1&sdkCorrelationID=84be61f08a829&storageID=uid_522fce5bad_mji6nda6mta&sessionID=uid_10589ebb1a_mdy6mzg6mdk&buttonSessionID=uid_d5eaa9f144_mdy6mzk6mjc&env=local&fundingEligibility=eyJwYXlwYWwiOnsiZWxpZ2libGUiOnRydWUsInZhdWx0YWJsZSI6dHJ1ZX0sInBheWxhdGVyIjp7ImVsaWdpYmxlIjpmYWxzZSwicHJvZHVjdHMiOnsicGF5SW40Ijp7ImVsaWdpYmxlIjpmYWxzZSwidmFyaWFudCI6bnVsbH0sInBheWxhdGVyIjp7ImVsaWdpYmxlIjpmYWxzZSwidmFyaWFudCI6bnVsbH19fSwiY2FyZCI6eyJlbGlnaWJsZSI6dHJ1ZSwiYnJhbmRlZCI6ZmFsc2UsImluc3RhbGxtZW50cyI6ZmFsc2UsInZlbmRvcnMiOnsidmlzYSI6eyJlbGlnaWJsZSI6dHJ1ZSwidmF1bHRhYmxlIjp0cnVlfSwibWFzdGVyY2FyZCI6eyJlbGlnaWJsZSI6dHJ1ZSwidmF1bHRhYmxlIjp0cnVlfSwiYW1leCI6eyJlbGlnaWJsZSI6dHJ1ZSwidmF1bHRhYmxlIjp0cnVlfSwiZGlzY292ZXIiOnsiZWxpZ2libGUiOnRydWUsInZhdWx0YWJsZSI6dHJ1ZX0sImhpcGVyIjp7ImVsaWdpYmxlIjpmYWxzZSwidmF1bHRhYmxlIjpmYWxzZX0sImVsbyI6eyJlbGlnaWJsZSI6ZmFsc2UsInZhdWx0YWJsZSI6dHJ1ZX0sImpjYiI6eyJlbGlnaWJsZSI6ZmFsc2UsInZhdWx0YWJsZSI6dHJ1ZX19fSwidmVubW8iOnsiZWxpZ2libGUiOnRydWV9LCJpdGF1Ijp7ImVsaWdpYmxlIjpmYWxzZX0sImNyZWRpdCI6eyJlbGlnaWJsZSI6dHJ1ZX0sImFwcGxlcGF5Ijp7ImVsaWdpYmxlIjpmYWxzZX0sInNlcGEiOnsiZWxpZ2libGUiOmZhbHNlfSwiaWRlYWwiOnsiZWxpZ2libGUiOmZhbHNlfSwiYmFuY29udGFjdCI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJnaXJvcGF5Ijp7ImVsaWdpYmxlIjpmYWxzZX0sImVwcyI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJzb2ZvcnQiOnsiZWxpZ2libGUiOmZhbHNlfSwibXliYW5rIjp7ImVsaWdpYmxlIjpmYWxzZX0sInAyNCI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJ6aW1wbGVyIjp7ImVsaWdpYmxlIjpmYWxzZX0sIndlY2hhdHBheSI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJwYXl1Ijp7ImVsaWdpYmxlIjpmYWxzZX0sImJsaWsiOnsiZWxpZ2libGUiOmZhbHNlfSwidHJ1c3RseSI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJveHhvIjp7ImVsaWdpYmxlIjpmYWxzZX0sIm1heGltYSI6eyJlbGlnaWJsZSI6ZmFsc2V9LCJib2xldG8iOnsiZWxpZ2libGUiOmZhbHNlfSwibWVyY2Fkb3BhZ28iOnsiZWxpZ2libGUiOmZhbHNlfX0&platform=desktop&remembered.0=venmo&experiment.enableVenmo=false&flow=purchase&currency=USD&intent=capture&commit=true&vault=false&enableFunding.0=venmo&debug=true&applePaySupport=false&supportsPopups=true&supportedNativeBrowser=false
```

image

## Related PR's

- https://github.com/paypal/paypal-identity-components/pull/14
- https://github.com/paypal/paypal-checkout-components/pull/1637
- https://github.com/paypal/paypal-sdk-client/pull/79
- https://github.paypal.com/Checkout/xo-content/pull/1747

### Important:

- [This needs to be merged before](https://github.paypal.com/Checkout/xo-content/pull/1723)

### CODE REVIEW

File: dist/module/locale.js [hunk 1]
```diff
@@ -319,7 +319,7 @@ export var COUNTRY_LANGS = {
   GT: [LANG.ES, LANG.EN, LANG.FR, LANG.ZH],
   GW: [LANG.EN, LANG.FR, LANG.ES, LANG.ZH],
   GY: [LANG.EN, LANG.FR, LANG.ES, LANG.ZH],
-  HK: [LANG.EN, LANG.ZH],
+  HK: [LANG.EN, LANG.ZH, LANG.ZH_HANT],
```
Comment (mimckenna) [inline dist/module/locale.js:322]: Add `LANG.ZH_HANT` to `TW` as well.  Taiwan only uses Traditional Chinese and has the same issue.

File: src/locale.js [hunk 2]
```diff
@@ -533,7 +534,7 @@ export const COUNTRY_LANGS : CountryMap = {
     GT: [ LANG.ES, LANG.EN, LANG.FR, LANG.ZH ],
     GW: [ LANG.EN, LANG.FR, LANG.ES, LANG.ZH ],
     GY: [ LANG.EN, LANG.FR, LANG.ES, LANG.ZH ],
-    HK: [ LANG.EN, LANG.ZH ],
+    HK: [ LANG.EN, LANG.ZH, LANG.ZH_HANT ],
```
Comment (mimckenna) [inline src/locale.js:537]: Add `LANG.ZH_HANT`  to `TW` as well.

**Review by mimckenna — CHANGES_REQUESTED** [review]
Please be sure `LANG.ZH_HANT` is supported for `TW`

File: dist/module/locale.js (hunk 1 above)
Comment (gabrielo91) [reply dist/module/locale.js:322]: Added!

File: src/locale.js (hunk 2 above)
Comment (gabrielo91) [reply src/locale.js:537]: done!

File: dist/module/locale.js [hunk 3]
```diff
@@ -427,7 +427,7 @@ export var COUNTRY_LANGS = {
   TR: [LANG.TR, LANG.EN],
   TT: [LANG.EN, LANG.FR, LANG.ES, LANG.ZH],
   TV: [LANG.EN, LANG.FR, LANG.ES, LANG.ZH],
-  TW: [LANG.ZH, LANG.EN],
+  TW: [LANG.ZH, LANG.EN, LANG.ZH_HANT],
```
Comment (mimckenna) [inline dist/module/locale.js:430]: This looks OK, although the first value in the list will be the default, which will be Simplified Chinese if no other context attribute is found.  Do you want `LANG.ZH_HANT` to be first, before `LANG.ZH` ?

File: src/locale.js [hunk 4]
```diff
@@ -641,7 +642,7 @@ export const COUNTRY_LANGS : CountryMap = {
     TR: [ LANG.TR, LANG.EN ],
     TT: [ LANG.EN, LANG.FR, LANG.ES, LANG.ZH ],
     TV: [ LANG.EN, LANG.FR, LANG.ES, LANG.ZH ],
-    TW: [ LANG.ZH, LANG.EN ],
+    TW: [ LANG.ZH, LANG.EN, LANG.ZH_HANT ],
```
Comment (mimckenna) [inline src/locale.js:645]: Ditto on the order of the constants.

File: dist/module/locale.js (hunk 3 above)
Comment (gabrielo91) [reply dist/module/locale.js:430]: Sure! Let me make the change!

**Comment by gabrielo91** [comment]
@mimckenna changes implemented! thanks! I just tried and it seems work fine, I just added some screenshots

**Review by mimckenna — APPROVED** [review]
This looks great!  I'm happy 👍

**Review by gregjopa — APPROVED** [review]

**Pull request merged**